frb/surveys/wise.py

```diff
@@ -0,0 +1,116 @@
+"""WISE Survey"""
+
+from IPython import embed
+
+import numpy as np
+from astropy import units, io, utils
+
+from frb.surveys import dlsurvey
+from frb.surveys import catalog_utils
+
+# Dependencies
+try:
+    from pyvo.dal import sia
+except ImportError:
+    print("Warning:  You need to install pyvo to retrieve DES images")
+    _svc = None
+else:
+    _DEF_ACCESS_URL = "https://datalab.noao.edu/sia/des_dr1"
+    _svc = sia.SIAService(_DEF_ACCESS_URL)
+
+# Define the data model for DES data
+photom = {}
+
+# DES-WISE
+photom['WISE'] = {}
+DES_WISE_bands = ['W1', 'W2', 'W3', 'W4']
+for band in DES_WISE_bands:
+    photom['WISE']['{:s}'.format(band)] = '{:s}mpro'.format(band.lower())
+    photom['WISE']['{:s}_err'.format(band)] = '{:s}sigmpro'.format(band.lower())
+photom['WISE']['ra'] = 'ra'
+photom['WISE']['dec'] = 'dec'
+
+
+class WISE_Survey(dlsurvey.DL_Survey):
+    """
+    Class to handle queries on the WISE survey
+
+    Child of DL_Survey which uses datalab to access NOAO
+
+    Args:
+        coord (SkyCoord): Coordiante for surveying around
+        radius (Angle): Search radius around the coordinate
+
+    """
+
+    def __init__(self, coord, radius, **kwargs):
+        dlsurvey.DL_Survey.__init__(self, coord, radius, **kwargs)
+        self.survey = 'WISE'
+        self.bands = ['W1', 'W2', 'W3', 'W4']
+        self.svc = sia.SIAService("https://datalab.noao.edu/sia/allwise")
+        self.qc_profile = "default"
+        self.database = "allwise.source"
+
+    def get_catalog(self, query=None, query_fields=None, print_query=False):
+        """
+        Grab a catalog of sources around the input coordinate to the search radius
+
+        Args:
+            query: Not used
+            query_fields (list, optional): Over-ride list of items to query
+            print_query (bool): Print the SQL query generated
+
+        Returns:
+            astropy.table.Table:  Catalog of sources returned.  Includes WISE
+            photometry for matched sources.
+        """
+        # Main DES query
+        main_cat = super(WISE_Survey, self).get_catalog(query_fields=query_fields, print_query=print_query)
+        if len(main_cat) == 0:
+            return main_cat
+        main_cat = catalog_utils.clean_cat(main_cat, photom['WISE'], fill_mask=-999.)
+
+        # Finish
+        self.catalog = main_cat
+        self.validate_catalog()
+        return self.catalog
+
+    def _gen_cat_query(self,query_fields=None):
+        """
+        Generate SQL query for catalog search
+
+        self.query is modified in place
+
+        Args:
+            query_fields (list):  Override the default list for the SQL query
+
+        """
+        if query_fields is None:
+            object_id_fields = ['source_id','ra','dec']
+            mag_fields = ['w1mpro', 'w2mpro', 'w3mpro', 'w4mpro',
```

Comment:
Three flags could be added to the default fields: the photometric quality flag `ph_qual`, the scattered moonlight contamination flag `moon_lev` and `tmass_key`, the cross matched source ID in the 2MASS point source catalog. Might help in processing the data later.

Reply:
added